src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpSyntaxNode.cs

```diff
@@ -48,15 +59,34 @@ internal abstract class HttpSyntaxNode : HttpSyntaxNodeOrToken
 
     private void GrowSpan(HttpSyntaxNodeOrToken child)
     {
-        if (_span == default)
+        if (_fullSpan == default)
         {
+            _fullSpan = child.FullSpan;
             _span = child.Span;
         }
         else
         {
-            var _spanStart = Math.Min(_span.Start, child.Span.Start);
-            var _spanEnd = Math.Max(_span.End, child.Span.End);
-            _span = new TextSpan(_spanStart, _spanEnd - _span.Start);
+            var fullSpanStart = Math.Min(_fullSpan.Start, child.FullSpan.Start);
+            var fullSpanEnd = Math.Max(_fullSpan.End, child.FullSpan.End);
+            _fullSpan = new TextSpan(fullSpanStart, fullSpanEnd - _fullSpan.Start);
+
+            var firstSignificantNodeOrToken = ChildNodesAndTokens
+                .FirstOrDefault(n => n.IsSignificant);
+
+            var lastSignificantNodeOrToken = ChildNodesAndTokens
+                .LastOrDefault(n => n.IsSignificant);
```

Comment:
I think we can improve this part by remembering the first and last significant children in fields as children are added inside `Add` below. That way we won't need to traverse all the child nodes here each time a child is added.